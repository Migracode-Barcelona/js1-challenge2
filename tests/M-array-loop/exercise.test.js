// Import the function to test
const { logDaysStartingWithT } = require("../../M-array-loop/exercise.js");

// Describe the test suite
describe("logDaysStartingWithT", () => {
  // Test case 1
  it("logs 'Tuesday' and 'Thursday'", () => {
    const days = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"];

    // Capture console output
    const consoleSpy = jest.spyOn(console, "log").mockImplementation(() => {});
    
    // Call the function being tested
    logDaysStartingWithT(days);

    // Assert that console.log output matches the snapshot
    expect(console.log.mock.calls).toMatchSnapshot();

    // Clean up
    consoleSpy.mockRestore();
  });

  // Test case 2
  it("logs no days", () => {
    const days = ["Monday", "Wednesday", "Friday", "Saturday", "Sunday"];

    // Capture console output
    const consoleSpy = jest.spyOn(console, "log").mockImplementation(() => {});
    
    // Call the function being tested
    logDaysStartingWithT(days);

    // Assert that console.log output matches the snapshot
    expect(console.log.mock.calls).toMatchSnapshot();

    // Clean up
    consoleSpy.mockRestore();
  });
});
