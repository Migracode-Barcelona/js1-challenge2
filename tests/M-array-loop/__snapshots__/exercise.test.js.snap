// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logDaysStartingWithT logs 'Tuesday' and 'Thursday' 1`] = `
[
  [
    "Tuesday",
  ],
  [
    "Thursday",
  ],
]
`;

exports[`logDaysStartingWithT logs no days 1`] = `[]`;
